lib/SPIRV/libSPIRV/SPIRVInstruction.h

```diff
@@ -2044,10 +2044,32 @@ _SPIRV_OP(GroupCommitReadPipe, false, 6)
 _SPIRV_OP(GroupCommitWritePipe, false, 6)
 #undef _SPIRV_OP
 
-class SPIRVAtomicInstBase:public SPIRVInstTemplateBase {
+class SPIRVAtomicInstBase : public SPIRVInstTemplateBase {
 public:
   SPIRVCapVec getRequiredCapability() const override {
-    return getVec(CapabilityInt64Atomics);
+    SPIRVCapVec CapVec;
+    // Most of atomic instructions do not require any capabilities
+    // ... unless they operate on 64-bit integers.
+    if (hasType() && getType()->isTypeInt(64)) {

```

Comment:
Probably just to simplify the implementation we can process both cases in setOpWords method, where we have all information we need to decide whether we need Int64Atomics capability or not.